lib/vmpooler/api/v1.rb

```diff
@@ -375,8 +390,16 @@ def atomically_allocate_vms(payload)
       content_type :json
       result = { 'ok' => false }
 
-      if jdata and !jdata.empty?
-        result = atomically_allocate_vms(jdata)
+      if jdata
+        empty = jdata.delete('empty')
+        invalid = jdata.delete('invalid')
+        statsd.increment(statsd_prefix + '.checkout.empty', empty) if !empty.nil?
+        statsd.increment(statsd_prefix + '.checkout.invalid', invalid) if !invalid.nil?
+        if !jdata.empty?
```

Comment:
I "overloaded" the jdata stuff to included the empty/invalid bits.  So this is saying, after taking those bits out see if the hash is empty or not to decide if we shall proceed.